test/conformance/protocol_test.go

```diff
@@ -70,16 +70,13 @@ func (pt *protocolsTest) makeRequest(domain string) *spoof.Response {
 
 	resp, err := pkgTest.WaitForEndpointState(
 		pt.clients.KubeClient, pt.t.Logf, domain,
-		pkgTest.Retrying(
-			func(resp *spoof.Response) (bool, error) {
-				if resp.StatusCode == http.StatusOK {
-					return true, nil
-				}
-
-				return true, fmt.Errorf("unexpected status: %d", resp.StatusCode)
-			},
-			http.StatusNotFound,
-		),
+		test.RetryingRouteInconsistency(func(resp *spoof.Response) (bool, error) {
+			if resp.StatusCode == http.StatusOK {
```

Comment:
One more nit:
I think this one could be shortened to `pkgTest.IsStatusOK()`.

Reply:
That's addressed by #3259 (which should've gone in already if it wasn't for the flakes). I'd keep it there to keep the changes scoped as they were planned to be.